tchannel/sync/client.py

```diff
@@ -52,131 +45,49 @@ class TChannelSyncClient(object):
 
     """
 
-    def __init__(self, name, process_name=None, known_peers=None, trace=False):
+    def __init__(
+        self,
+        name,
+        hostport=None,
+        process_name=None,
+        known_peers=None,
+        trace=False,
+        threadloop=None,
+    ):
         """Initialize a new TChannelClient.
 
         :param process_name:
             Name of the calling process. Used for logging purposes only.
         """
-        self._async_client = async.TChannel(
+        super(TChannelSyncClient, self).__init__(
             name,
+            hostport=hostport,
             process_name=process_name,
             known_peers=known_peers,
-            trace=trace
+            trace=trace,
+
         )
         self._threadloop = ThreadLoop()
         self._threadloop.start()
 
-    def request(self, *args, **kwargs):
-        """Initiate a new request to a peer.
-
-        :param hostport:
-            If specified, requests will be sent to the specific host.
-            Otherwise, a known peer will be picked at random.
-        :param service:
-            Name of the service being called. Defaults to an empty string.
-        :param service_threshold:
-            If ``hostport`` was not specified, this specifies the score
-            threshold at or below which peers will be ignored.
-        :returns SyncClientOperation:
-            An object with a ``send(arg1, arg2, arg3)`` operation.
-        """
-        operation = self._async_client.request(*args, **kwargs)
-        operation = SyncClientOperation(operation, self._threadloop)
-
-        return operation
-
-    def advertise(self, routers, name=None, timeout=None):
-        """Advertise with Hyperbahn.
-
-        :param routers: list of hyperbahn addresses to advertise to.
-        :param name: service name to advertise with.
-        :param timeout: backoff period for failed requests.
-        :returns: first advertise result.
-        :raises AdvertiseError: when unable to begin advertising.
-        """
-
-        @gen.coroutine
-        def make_request():
-
-            response = yield self._async_client.advertise(
-                routers=routers,
-                name=name,
-                timeout=timeout,
-            )
-
-            header = yield response.get_header()
-            body = yield response.get_body()
-
-            result = Response(header, body)
-
-            raise gen.Return(result)
+        self.advertise = self._wrap(self.advertise)
 
-        future = self._threadloop.submit(make_request)
+        self.raw = self._wrap(self.raw)
+        self.thrift = self._wrap(self.thrift)
+        self.json = self._wrap(self.json)
```

Comment:
Overriding `call` should be enough since all the argscheme's use `call` to ultimately make the request.